### DISCUSSION TITLE
Webpack module federation throwing `fn is not a function ` when trying to load a remote from a vite exposed component

### DISCUSSION THREAD

**Discussion by OwenPattison** [webpack/webpack#18353] [Q&A]
I have been attempting to load vue 3 components from a vite micro-frontend using a webpack host that runs vue 2, my main goal is to bridge the gap between vue versions and use this as a migration path.

I have been running into issues when trying to render the remote component which is where I've been seeing this error:
![image](https://github.com/webpack/webpack/assets/5864244/cc602551-b28e-4f62-a6b4-45c88061fe0e)

I have also made a *noddy* repo which replicates the issue I see in my solution which can be found [here](https://github.com/OwenPattison/federation-test)

As far as I can tell this should be possible given I have seen an example of it [here](https://github.com/originjs/vite-plugin-federation/tree/main/packages/examples/vue3-demo-webpack-esm-esm)

I've attempted many of the suggestions in various stackoverflows and github issues, this one resembled my issue the closest I would say: https://github.com/webpack/webpack/issues/12281 however I'm still unable to get this functioning correctly, any pointers would be appreciated.

**Comment by alexander-akait** [comment]
Hello, do you still need a help, sounds like vite module federatin plugin is broken...

**Reply by OwenPattison** [reply]
@alexander-akait Thanks for checking in. 
I have found an alternate solution to the issue, switching all the library types from `var` to `module`. 
[This is the line that seems to cause the issue](https://github.com/OwenPattison/federation-test/blob/main/host-app/vue.config.js#L10) using `module` in my host and remotes seems to be a good workaround - though I'm sure this issue will still affect others who may be unable to switch library type.
